### PR TITLE
feat(batchDelete): update endpoint to queue list items for batch delete

### PR DESCRIPTION
**Pull request by jpetto**
## Goal

update the as of yet unused `deleteShareableListItems` endpoint to instead query and queue items to be deleted.

- use @pocket-tools ts config

## Tickets

- https://getpocket.atlassian.net/browse/OSL-562

## Implementation Decisions

following the pattern established in [annotations API](https://github.com/Pocket/annotations-api/blob/main/src/server/routes/queueDelete.ts).

### CODE REVIEW

File: src/config/index.ts [hunk 1]
```diff
@@ -5,7 +5,11 @@ export default {
     defaultMaxAge: 0,
   },
   aws: {
-    region: process.env.AWS_DEFAULT_REGION || 'us-east-1',
+    endpoint:
+      process.env.NODE_ENV != 'production' &&
+      process.env.NODE_ENV != 'development'
+        ? process.env.AWS_ENDPOINT || 'http://localstack:4566'
```
Comment (katerinachinnappan) [inline src/config/index.ts:11]: should the port be `4567` and expose it in docker-compose as well? I think S3 endpoint is pointing to `4566`

File: src/config/index.ts [hunk 2]
```diff
@@ -18,6 +22,19 @@ export default {
         },
       },
     },
+    region: process.env.AWS_DEFAULT_REGION || 'us-east-1',
+    sqs: {
+      batchSize: 10,
+      listItemsDeleteQueue: {
+        url:
+          process.env.SQS_BATCH_LIST_ITEM_DELETE_QUEUE_URL ||
+          'http://localhost:4566/queue/ShareableListsApi-Local-Batch-Delete-Consumer-Queue',
```
Comment (katerinachinnappan) [inline src/config/index.ts:31]: I think port should be `4567`?

File: src/config/index.ts [hunk 3]
```diff
@@ -18,6 +22,19 @@ export default {
...
+      },
+    },
+  },
+  queueDeleteShareableListItems: {
+    queryLimit: 500,
+    externalIdChunkSize: 25,
```
Comment (katerinachinnappan) [inline src/config/index.ts:37]: so did we decide the limit is 25 items to delete?

File: src/express.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ import { getPublicContext, IPublicContext } from './public/context';
 import { getAdminContext, IAdminContext } from './admin/context';
 import { startAdminServer } from './admin/server';
 import deleteUserDataRouter from './public/routes/deleteUserData';
-import deleteShareableListItemsRouter from './public/routes/deleteShareableListItems';
+import deleteShareableListItemsRouter from './public/routes/queueDeleteShareableListItems';
```
Comment (katerinachinnappan) [inline src/express.ts:13]: nit pick - just for consistency should we rename this to `queueDeleteShareableListItemsRouter`

File: src/public/routes/queueDeleteShareableListItems.ts [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+import config from '../../config';
+import { client } from '../../database/client';
+import { serverLogger } from '../../express';
+import { sqs } from '../../aws/sqs';
+
+const router = Router();
+const db = client();
+
+// the body of the sqs message, essentially
+export type DeleteListItemsSqsMessage = {
```
Comment (katerinachinnappan) [inline src/public/routes/queueDeleteShareableListItems.ts:21]: again just a nit pick  -  rename for consistency to `QueueDeleteShareableListItemsSQSMessage`

File: src/public/routes/queueDeleteShareableListItems.ts [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+  // keeping userId and url in the message for logging purposes.
+  // may not be necessary, but it's a small amount of data that could be
+  // very helpful in debugging...
+  userId: number;
+  url: string;
+  externalIds: string[];
+};
+
+const deleteShareableListItemDataSchema: Schema = {
```
Comment (katerinachinnappan) [inline src/public/routes/queueDeleteShareableListItems.ts:30]: suggestion to rename to `queueDeleteShareableListItemDataSchema`

**Review by katerinachinnappan — CHANGES_REQUESTED** [review]

File: src/config/index.ts (hunk 1 above)
Comment (jpetto) [reply src/config/index.ts:11]: i think this is our first usage of an AWS endpoint in code? if so, it should be fine as-is. (also pretty sure localstack exposes _all_ services on port `4566` now.)

File: docker-compose.yml [hunk 7]
```diff
@@ -18,18 +18,22 @@ services:
...
+    # (fwiw localstack/localstack:latest also lists the above as "2.1.1.dev",
+    # so it doesn't appear that means it's a "dev" build)
+    # if you change the SHA below, please update build version comment above 😅
     image: localstack/localstack@sha256:16ffdd3c4ae5669ddb10f113f0527024723333b5a8e41ff43a2130e22d95e8d6
     env_file:
       - .docker/local.env
     environment:
       - LOCALSTACK_API_KEY=${LOCALSTACK_API_KEY}
       - DEBUG=1
-      - SERVICES=s3
```
Comment (jpetto) [inline docker-compose.yml:32]: services aren't required to be defined in localstack 2.1 😄 - they are spun up on demand.

**Review by katerinachinnappan — APPROVED** [review]
🚀

**Pull request merged**